### ISSUE TITLE
Consumers sometimes think that label is superfluous when using IconButton 

### ISSUE THREAD

**Issue by dougmacknz** [cultureamp/kaizen-design-system#1530]
I am seeing quite a few cases where people are using `<IconButton>` with an empty string for a label. I am hypothesizing  that people think it's pointless given it doesn't display visually, not realising the accessibility implications of providing no label.

I see a couple of things we could do to help this situation:
* ~Warn consumers in console when providing an empty string.~
* Rename the `label` prop to `accessibleLabel` (or similar) for `IconButton` to make it a bit more clear why you still need provide a label.

**UPDATE:** We've noticed that the similar console warnings for the Icon component haven't really worked in fixing the same issue over there (there are many warnings out there not being addressed). We need something better, and Jason has prototyped a way to force a non empty string below.

**Comment by jasononeil** [comment]
I found a typescript hack that might work ([Playground link](https://www.typescriptlang.org/play?jsx=1#code/JYWwDg9gTgLgBAIgEoFMCGBjGC4DMoQiJTpYIDcAsAFA0wCeYKcAchAHYCi4DAyjFGDsA5gB4AKnBQAPGCnYATAM5wlAocIB8cALxxJMuYpUByE3AD8cdigBuKKHABc+qrWoMmcAAoEwS0V4pWXllVXURbT0Abxo4OAAbNAAjFASXNi4een5BEUDNGgBfGhpcAFd2LGAOOAAxaAwUTO4wBgAZFLTA4KMwtTytAAowPyUXXwh-AoBKOFjqeJIYcqh2OFEwTVEAei3i0uoGqCaW7M7UhKHopMuXBAAJNISIBCKZmmPTjlaOrqubv8XOxygkEu9Po1mj9zv9rrc0vc3h93BgOGo4AArJTSACMug2X2hWTa9AuaUS-x0j2erzgO0K1DR7Ax2OkACYCaIiWdSeSEvTGczWTiAMxcnkwvn-SmXak4BlAA))

```
import "React" from "react";

type NonEmptyString<T extends string> = T extends '' ? never : T;

type Props<S extends string> = {
  label: NonEmptyString<S>
}

function ForceNonEmptyLabel<S extends string>(props: Props<S>) {
  return <p></p>
}

ForceNonEmptyLabel({label: "Hello"})
ForceNonEmptyLabel({label: null}) // error
ForceNonEmptyLabel({label: ""}) // error

const jsx1 = <ForceNonEmptyLabel label="Hello" />
const jsx2 = <ForceNonEmptyLabel /> // error
const jsx3 = <ForceNonEmptyLabel label="" /> // error
```

The error message isn't nice:

> Type string is not assignable to type never

but it works 🤷
